src/Microsoft.ML.LightGBM.StaticPipe/LightGbmStaticExtensions.cs

```diff
@@ -196,17 +307,57 @@ public static (Vector<float> score, Key<uint, TVal> predictedLabel)
             int? minDataPerLeaf = null,
             double? learningRate = null,
-            int numBoostRound = LightGbmArguments.Defaults.NumBoostRound,
-            Action<LightGbmArguments> advancedSettings = null,
+            int numBoostRound = Options.Defaults.NumBoostRound,
             Action<OvaModelParameters> onFit = null)
         {
-            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, advancedSettings, onFit);
+            CheckUserValues(label, features, weights, numLeaves, minDataPerLeaf, learningRate, numBoostRound, onFit);
 
             var rec = new TrainerEstimatorReconciler.MulticlassClassifier<TVal>(
                 (env, labelName, featuresName, weightsName) =>
                 {
                     var trainer = new LightGbmMulticlassTrainer(env, labelName, featuresName, weightsName, numLeaves,
-                       minDataPerLeaf, learningRate, numBoostRound, advancedSettings);
+                       minDataPerLeaf, learningRate, numBoostRound);
+
+                    if (onFit != null)
+                        return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                    return trainer;
+                }, label, features, weights);
+
+            return rec.Output;
+        }
+
+        /// <summary>
+        /// Predict a target using a tree multiclass classification model trained with the <see cref="LightGbmMulticlassTrainer"/>.
+        /// </summary>
+        /// <param name="catalog">The multiclass classification catalog trainer object.</param>
+        /// <param name="label">The label, or dependent variable.</param>
+        /// <param name="features">The features, or independent variables.</param>
+        /// <param name="weights">The weights column.</param>
+        /// <param name="options">Advanced options to the algorithm.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained. Note that this action cannot change the
+        /// result in any way; it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The set of output columns including in order the predicted per-class likelihoods (between 0 and 1, and summing up to 1), and the predicted label.</returns>
+        public static (Vector<float> score, Key<uint, TVal> predictedLabel)
+            LightGbm<TVal>(this MulticlassClassificationCatalog.MulticlassClassificationTrainers catalog,
+            Key<uint, TVal> label,
+            Vector<float> features,
+            Scalar<float> weights,
+            Options options,
+            Action<OvaModelParameters> onFit = null)
```

Comment:
Why `OVA`? Is this just one-versus-all under the hood?

Reply:
For multiclass,  LightGBM uses the OvaModelParameters

We can also see it in the class definition as well

[https://github.com/dotnet/machinelearning/blob/620ca893a2545ed57f65c324c9c26e171ffc4839/src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs#L25-L27](https://github.com/dotnet/machinelearning/blob/620ca893a2545ed57f65c324c9c26e171ffc4839/src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs#L25-L27)


---
In reply to: [251534510](https://github.com/dotnet/machinelearning/pull/2261#discussion_r251534510) [](ancestors = 251534510)